go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -960,14 +1007,18 @@ func (h *defaultNetworkControllerEventHandler) UpdateResource(oldObj, newObj int
 func (h *defaultNetworkControllerEventHandler) DeleteResource(obj, cachedObj interface{}) error {
 	switch h.objType {
 	case factory.PodType:
-		var portInfo *lpInfo
 		pod := obj.(*kapi.Pod)
 
-		if cachedObj != nil {
-			portInfo = cachedObj.(*lpInfo)
+		if h.oc.isPodScheduledinLocalZone(pod) {
```

Comment:
not sure if `scheduled` should be checked here. We need this logic to be able to remove zonbie pods, that are in the local zone but never got scheduled, right?

Reply:
good point @flavio-fernandes ! the problem is @numansiddique 's logic of "is a pod in local zone or not" depends on the pod.nodeName being set. -> so unless the nodeName is set the zombie pod would be in the land of nowhere ? :D

However I am also now thinking, can we have situations where pod isn't scheduled and stuck in scheduling? in which case are we just ok to not process it? Should be fine I guess...